code/modules/power/apc.dm

```diff
@@ -151,20 +168,31 @@
 	if(terminal)
 		terminal.connect_to_network()
 
-/obj/machinery/power/apc/New(turf/loc, direction, building = 0)
+/obj/machinery/power/apc/New(turf/loc, ndir, building = 0)
 	if(!armor)
 		armor = list("melee" = 20, "bullet" = 20, "laser" = 10, "energy" = 100, "bomb" = 30, "bio" = 100, "rad" = 100, "fire" = 90, "acid" = 50)
 	..()
 	GLOB.apcs += src
 	GLOB.apcs = sortAtom(GLOB.apcs)
 
 	wires = new(src)
-
+	// offset 24 pixels in direction of dir
+	// this allows the APC to be embedded in a wall, yet still inside an area
+	if(building)
+		setDir(ndir)
+	tdir = dir // to fix Vars bug
```

Comment:
Can't this just be
```suggestion
	tdir = ndir // to fix Vars bug
```
Since you're setting `dir` to `SOUTH` on the next line. Also what is the "vars bug"?

Reply:
This is mostly for record keeping; if you set dir manually, you're going to wind up with oddities between it being set, at round start for pre-made ones, and ones that you actually build.